src/Domain/CoordinateMaps/ProductMaps.hpp

```diff
@@ -21,19 +21,19 @@ namespace product_detail {
 template <typename T, size_t Size, typename Map1, typename Map2,
           typename Function, size_t... Is, size_t... Js>
 std::array<std::decay_t<tt::remove_reference_wrapper_t<T>>, Size> apply_call(
-    const std::array<T, Size>& xi, const Map1& map1,
+    const std::array<T, Size>& source_coords, const Map1& map1,
```

Comment:
This function is used with both `source_coords` and `target_coords` (for `operator()` and `inverse`, respectively).  Maybe just `coords`?